libc/stdio_printf.c

```diff
@@ -0,0 +1,3582 @@
+		int stdout_fd = dup(STDOUT_FILENO); \
+		int fd = open(PATH, O_WRONLY | O_CREAT, 0666); \
+		dup2(fd, STDOUT_FILENO); \
+		printf(format, value); \
+		fflush(stdout); \
+		fseek(output_file, 0, SEEK_SET); \
+		fflush(output_file); \
+		/* redirect back to stdout */ \
+		dup2(stdout_fd, STDOUT_FILENO); \
+		close(fd); \
+		close(stdout_fd); \
+		TEST_ASSERT_EQUAL_STRING(expect, fgets(buff, strlen(expect) + 1, output_file)); \
+		TEST_ASSERT_EQUAL_STRING(expect, buff); \
+	}
+
+#define sntest(format, value, expect) \
+	{ \
+		memset(buff, 0, sizeof(buff)); \
+		snprintf(buff, strlen(expect) + 1, format, value); \
+		TEST_ASSERT_EQUAL_STRING(expect, buff); \
+	}
+
+#define stest(format, value, expect) \
+	{ \
+		memset(buff, 0, sizeof(buff)); \
+		sprintf(buff, format, value); \
+		TEST_ASSERT_EQUAL_STRING(expect, buff); \
+	}
+
+/* wrappers for vprintf tests*/
+static void vdtest(char *expect, const char *format, ...)
+{
+	va_list arg_ptr;
+
+	memset(buff, 0, sizeof(buff));
+	rewind(output_file);
+
+	va_start(arg_ptr, format);
+
+	vdprintf(fileno(output_file), format, arg_ptr);
+	fseek(output_file, 0, SEEK_SET);
+	fflush(output_file);
+
+	TEST_ASSERT_EQUAL_STRING(expect, fgets(buff, strlen(expect) + 1, output_file));
+	TEST_ASSERT_EQUAL_STRING(expect, buff);
+
+	va_end(arg_ptr);
+}
+
+
+static void vftest(char *expect, const char *format, ...)
+{
+	va_list arg_ptr;
+
+	memset(buff, 0, sizeof(buff));
+	rewind(output_file);
+
+	va_start(arg_ptr, format);
+
+	vfprintf(output_file, format, arg_ptr);
+	fseek(output_file, 0, SEEK_SET);
+	fflush(output_file);
+
+	TEST_ASSERT_EQUAL_STRING(expect, fgets(buff, strlen(expect) + 1, output_file));
+	TEST_ASSERT_EQUAL_STRING(expect, buff);
+
+	va_end(arg_ptr);
+}
+
+
+static void vtest(char *expect, const char *format, ...)
+{
+	va_list arg_ptr;
+
+	rewind(output_file);
+	int stdout_fd = dup(STDOUT_FILENO);
+	memset(buff, 0, sizeof(buff));
+	int fd = open(PATH, O_WRONLY | O_CREAT, 0666);
+	dup2(fd, STDOUT_FILENO);
+
+	va_start(arg_ptr, format);
+
+	vprintf(format, arg_ptr);
+
+	fflush(stdout);
+	fseek(output_file, 0, SEEK_SET);
+	fflush(output_file);
+	/* redirect back to stdout */
+	dup2(stdout_fd, STDOUT_FILENO);
+	close(fd);
+	close(stdout_fd);
+	TEST_ASSERT_EQUAL_STRING(expect, fgets(buff, strlen(expect) + 1, output_file));
+	TEST_ASSERT_EQUAL_STRING(expect, buff);
+
+	va_end(arg_ptr);
+}
+
+
+static void vsntest(char *expect, const char *format, ...)
+{
+	va_list arg_ptr;
+	memset(buff, 0, sizeof(buff));
+
+	va_start(arg_ptr, format);
+
+	vsnprintf(buff, strlen(expect) + 1, format, arg_ptr);
+	TEST_ASSERT_EQUAL_STRING(expect, buff);
+
+	va_end(arg_ptr);
+}
+
+
+static void vstest(char *expect, const char *format, ...)
+{
+	va_list arg_ptr;
+	memset(buff, 0, sizeof(buff));
+
+	va_start(arg_ptr, format);
+
+	vsprintf(buff, format, arg_ptr);
+	TEST_ASSERT_EQUAL_STRING(expect, buff);
+
+	va_end(arg_ptr);
+}
+
+
+/* other functions */
+static char *signed_to_string(long long int value, int base)
```

Comment:
Let's prefix functions names like that with `test_` and use camelCase as it's mentioned in our coding convention. I am not sure about renaming macros above, maybe let's leave them as they are to not increase binary size so much.

```suggestion
static char *test_signedToStr(long long int value, int base)
```

We could do it for the rest of functions analogically.